lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -34,32 +34,46 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel.
+	 * Get the presence state for this Channel. Implicitly attaches the
+	 * Channel. However, if the channel is in or moves to the FAILED
+	 * state before the operation succeeds, it will result in an error
 	 * @return: the current present members.
 	 * @throws AblyException
 	 */
-	public synchronized PresenceMessage[] get()  {
+	public synchronized PresenceMessage[] get() throws AblyException {
+		if (channel.state == ChannelState.failed) {
```

Comment:
If the channel is in the failed state, does the `attach()` throw an exception? If so, the check above (l44) can be removed.